Makefile

```diff
@@ -60,11 +63,19 @@ generate: controller-gen
 
 # Build the docker image
 docker-build: test
-	docker build . -t ${IMG}
+	docker build . -t ${IMAGE}:${DEV_VERSION}
+
+# Build the release docker image
+docker-build-release: test
```

Comment:
1. why don't build it to be REPO directly instead of retag it.
2. i don't think we should use 'latest" version here. it should be like v1.0.0

Reply:
a better logic might be 
1. avoid this build-release and push-release(instead only have this original docker-build & docker-push command).
2. specify image to be (AWS_ACCOUNT).dkr.ecr.$(AWS_REGION).amazonaws.com/amazon/appmesh-controller
3. use a single version instead of version and devVersion, which is result of "git describe --tags --dirty --always"
4. if we want to release a GA version, we just tag the commit like v1.0.0, and docker-build && docker-push
5. if we want to release a Dev version, we just override the IMAGE completely or just aws account.

Reply:
+1 on avoiding build-release & push-release and instead using existing docker-build and docker-push. I believe we had this build-release, push-release format in the old controller...

Reply:
I'm more inclined towards having explicit targets for whats being done rather than relying on the caller but both approaches are fine with the use of env. I will adjust this